go-controller/pkg/ovn/egressip.go

```diff
@@ -2442,13 +2485,78 @@ func (e *EgressIPController) deleteExternalGWPodSNATOps(ni util.NetInfo, ops []o
 	return ops, nil
 }
 
+// getLayerScopedNextHop determines the next hop for a given Node considering the network topology type
+func (e *EgressIPController) getLayerScopedNextHop(ni util.NetInfo, nodeName string, wantsIPv6 bool) (net.IP, error) {
+	if ni.TopologyType() == types.Layer3Topology {
+		return e.getGatewayRouterJoinIP(ni, nodeName, wantsIPv6)
+	}
+	if ni.TopologyType() == types.Layer2Topology {
+		// For L2, if the pod selected by an EgressIP and also that EgressIP is managed by this controller, then the next hop must be
+		// retrieved from the node annotation. If a node is managed by another controller, we must get the remote Node gateway router to switch port IP
+		// from the node annotation.
+		// FIXME: remove gathering the required information from a Node annotations as this approach does not scale
+		node, err := e.watchFactory.GetNode(nodeName)
+		if err != nil {
+			return nil, fmt.Errorf("faild to retrive Node %s: %w", nodeName, err)
+		}
+		localNode, err := e.getALocalZoneNodeName()
+		if err != nil {
+			return nil, err
+		}
+		// node manager by this controller
+		if localNode == nodeName {
+			nextHopIPs, err := util.ParseNodeL3GatewayAnnotation(node)
+			if err != nil {
+				if util.IsAnnotationNotSetError(err) {
+					// remote node may not have the annotation yet, suppress it
+					return nil, types.NewSuppressedError(err)
+				}
+				return nil, fmt.Errorf("failed to get the node %s L3 gateway annotation: %w", node.Name, err)
+			}
+			if len(nextHopIPs.NextHops) == 0 {
+				return nil, fmt.Errorf("annotation on the node %s had empty next hop IPs", node.Name)
+			}
+			return util.MatchFirstIPFamily(wantsIPv6, nextHopIPs.NextHops)
```

Comment:
I don't understand this. The next hop here will be the default gateway IP out of the cluster. IIUC this function is used to find the next hop to put as the reroute IP in the LRP. Why would you ever put the next hop outside of the cluster for that? Wouldn't you just not render the LRP at all and let the normal IP routing do its job?

Reply:
> I don't understand this.

That makes two of us :)
I think I wrote this when there was no static routes associated with the L2 gw router. Not sure.
Ill rework this.